### PR TITLE
Network service support on renderer

### PR DESCRIPTION
**Pull request by venumusham**
Network Service Implementation on Renderer

### CODE REVIEW

**Comment by venumusham** [comment]
WIP Commit, Do not Merge

File: base/memory/shared_memory_posix.cc [hunk 1]
```diff
@@ -221,6 +223,7 @@ bool SharedMemory::Create(const SharedMemoryCreateOptions& options) {
   readonly_shm_ =
       SharedMemoryHandle(base::FileDescriptor(readonly_mapped_file, false),
                          options.size, shm_.GetGUID(), shared_memory_file_id);
+ SetMemoryId(shared_memory_file_id);
```
Comment (joonghunpark) [inline base/memory/shared_memory_posix.cc]: It seems that this statement is not used anymore.

File: base/memory/shared_memory.h [hunk 2]
```diff
@@ -214,7 +214,10 @@ class BASE_EXPORT SharedMemory {
   // that was mapped. The ID is valid even after the SharedMemoryHandle is
   // Closed, as long as the region is not unmapped.
   const UnguessableToken& mapped_id() const { return mapped_id_; }
-
+#if defined(NETWORK_SHARED_MEMORY)
+  int GetMemoryId() const { return shared_memory_id_; }
+  void SetMemoryId(int id) { shared_memory_id_ = id; }
```
[the remark this answers is not in the document]
Comment (venumusham) [reply base/memory/shared_memory.h]: Is Get/SetMemoryId being used currently? =< Not used in current CL, will cleanup

**Comment by venumusham** [comment]
For New changes outside CASTANETS macros, change is related to adding additional parameter in Serialize API, default value is set to zero. hence we can keep the changes.

Verified builds a) without castanets, b) without Network file system.
No issues found with this CL.

**Pull request merged**